doc/lag/Static Port Channel Design Document.md

```diff
@@ -0,0 +1,278 @@
+### 1.2.1 Basic Approach
+A Static Port Channel has its member ports always available for traffic as long the links are up. All ports are mandated to have the same speed to become members of a Port Channel. Teamd uses round-robin for a Static Port Channel.
+
+### 1.2.2 Container
+An instance of Teamd process is created when a Port Channel (Static or Dynamic) is created. Teamd process runs as part of the Teamd docker. 
+
+### 1.2.3 SAI Overview
+There are no changes to SAI headers/implementation to support this feature. 
+
+# 2 Functionality
+## 2.1 Target Deployment Use Cases
+Static Port Channels can be deployed in both Enterprise and Data center networks. Example usage scenarios for Static Port Channels are:
+
+  - To attach legacy equipment that does not support LACP.
+  - To handle warm boot cases where the control plane outage can exceed 3 * long timeout.
+
+### 2.1.1 Enterprise
+![Enterprise](images/enterprise_lag.PNG)
+
+### 2.1.2 Data Center
+![Data Center](images/data_center_lag.PNG)
+
+## 2.2 Functional Description
+A Port Channel bundles individual ports into a group to provide increased bandwidth, load balancing and redundancy. A Port Channel can be either Static or Dynamic. A Dynamic Port Channel runs Link Aggregation Signaling protocols like LACP to mark the links active for traffic. A Static Port Channel has its member ports always available for traffic. Incoming LACPDUs are silently ignored.
+
+Static Port Channel uses the roundrobin runner which hashes CPU sourced or CPU forwarded traffic among the member ports. Traffic forwarded in hardware on the Port Channel uses the silicon specific traffic hashing algorithm. 
+
+# 3 Design
+
+## 3.1 Overview
+Upon creating a static Port Channel, the "Teamd" daemon corresponding to that Port Channel is instantiated in static mode.
+
+### 3.1.1 Teamd configuration
+A user should be able to create the Port Channel in static or dynamic mode. For static mode, the CONFIG_DB is updated with a "on": "true".
+Note: Port Channels are dynamic by default in SONiC. 
+
+### 3.1.2 Team Manager
+Team Manager listens to Port Channel update events from the PORTCHANNEL and PORTCHANNEL_MEMBER table in CONFIG_DB. For the Static Port Channel add event it starts Teamd  with round-robin as the runner; whereas for the dynamic Port Channel Teammgr starts Teamd with lacp as the runner. 
+
+When static port-channel is configured then teammgr invokes teamd instance as follows
+root          27       1  0 Jun17 ?        00:01:10 /usr/bin/teamd -r -t PortChannel0001 -c {"device":"PortChannel0001","hwaddr":"00:e0:ec:7a:88:11","runner":{"name":"roundrobin","min_ports":1}} -L /var/warmboot/teamd/ -g -d
+
+round-robin load-balancing is configured for the kernel interface port-channel.
+
+## 3.2 DB Changes
+### 3.2.1 CONFIG DB
+To support the static mode for a Port Channel, the PORTCHANNEL table is modified to add a new key-value pair where the value is a "true" or a "false".
+```
+"PORTCHANNEL|PortChannel0001": {
```

Comment:
Please change the yang model as required. If you need to impose restrictions between switching lacp vs static it has to be done through yang models too apart from CLI.

Reply:
Hi Dgsudharsan,

Its already present in openconfig-if-aggregate.yang

  typedef aggregation-type {
    type enumeration {
      enum LACP {
        description "LAG managed by LACP";
      }
      enum STATIC {
        description "Statically configured bundle / LAG";
      }
    }
    description
      "Type to define the lag-type, i.e., how the LAG is
      defined and managed";
  }

Thanks,
Kannan.S